src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -90,6 +91,45 @@ public void Emit(List<AssociativeNode> astList)
             asm.Save("DynamicAssembly.dll");
         }
 
+        internal Dictionary<string,IList> EmitAndExecute(List<AssociativeNode> astList)
+        {
+            //TODO refactor with above method.
+            compilePass = CompilePass.MethodLookup;
+            foreach (var ast in astList)
+            {
+                DfsTraverse(ast);
+            }
+
+            // 1. Create assembly builder (dynamic assembly)
+            var asm = BuilderHelper.CreateAssemblyBuilder("DynamicAssembly", false);
```

Comment:
I guess we could add another option here to `CreateAssemblyBuilder` to only create an assembly with `AssemblyBuilderAccess.Run` or `AssemblyBuilderAccess.RunAndCollect` option in this case. See https://github.com/DynamoDS/Dynamo/blob/4f5610ff713d90a842c691429648ef98ab3477c0/src/Engine/EmitMSIL/BuilderHelper.cs#L40.

Reply:
refactored this a bit.